Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -7247,6 +7265,2013 @@ def panorama_install_file_content_update_command(args: dict):
+    """
+    :param state: New HA State
+    """
+    state: str
+    _output_prefix = OUTPUT_PREFIX + "HAStateUpdate"
+    _title = "PAN-OS High-Availability Updated State"
+
+
+@dataclass
+class DownloadSoftwareCommandResult:
+    summary_data: List[GenericSoftwareStatus]
+    result_data: None = None
+
+    _output_prefix = OUTPUT_PREFIX + "DownloadStatus"
+    _title = "PAN-OS Software Download request Status"
+
+    _summary_cls = GenericSoftwareStatus
+
+
+@dataclass
+class InstallSoftwareCommandResult:
+    summary_data: List[GenericSoftwareStatus]
+    result_data: None = None
+
+    _output_prefix = OUTPUT_PREFIX + "InstallStatus"
+    _title = "PAN-OS Software Install request Status"
+
+    _summary_cls = GenericSoftwareStatus
+
+
+@dataclass
+class RestartSystemCommandResult:
+    summary_data: List[GenericSoftwareStatus]
+    result_data: None = None
+
+    _output_prefix = OUTPUT_PREFIX + "RestartStatus"
+    _title = "PAN-OS Software Restart request Status"
+
+    _summary_cls = GenericSoftwareStatus
+
+
+@dataclass
+class CheckSystemStatus(ResultData):
+    """
+    :param up: Whether the host device is up or still unavailable.
+    """
+    up: bool
+
+    _output_prefix = OUTPUT_PREFIX + "SystemStatus"
+    _title = "PAN-OS System Status"
+    _outputs_key_field = "hostid"
+
+
+@dataclass
+class DeviceGroupInformation(ResultData):
+    """
+    :param serial: Serial number of firewall
+    :param connected: Whether the firewall is currently connected
+    :param hostname: Firewall hostname
+    :param last_commit_all_state_sp: Text state of last commit
+    :param name: Device group Name
+    """
+    serial: str
+    connected: str
+    hostname: str
+    last_commit_all_state_sp: str
+    name: str = ""
+
+    _output_prefix = OUTPUT_PREFIX + "DeviceGroupOp"
+    _title = "PAN-OS Operational Device Group Status"
+    _outputs_key_field = "name"
+
+
+@dataclass
+class TemplateStackInformation(ResultData):
+    """
+    :param serial: Serial number of firewall
+    :param connected: Whether the firewall is currently connected
+    :param hostname: Firewall hostname
+    :param last_commit_all_state_tpl: Text state of last commit
+    :param name: Template Stack Name
+    """
+    serial: str
+    connected: str
+    hostname: str
+    last_commit_all_state_tpl: str
+    name: str = ""
+
+    _output_prefix = OUTPUT_PREFIX + "TemplateStackOp"
+    _title = "PAN-OS Operational Template Stack status"
+    _outputs_key_field = "name"
+
+
+@dataclass
+class PanosObjectReference(ResultData):
+    """
+    :param container_name: What parent container (DG, Template, VSYS) this object belongs to.
+    :param name: The PAN-OS object name
+    :param object_type: The PAN-OS-Python object type
+    """
+    container_name: str
+    name: str
+    object_type: str
+
+    _output_prefix = OUTPUT_PREFIX + "PanosObject"
+    _title = "PAN-OS Objects"
+
+
+def dataclass_from_dict(device: Union[Panorama, Firewall], d: dict, class_type: Callable):
```

Comment:
I suggest to give a more meaningful names here to ```d``` and to ```class_type```. 
any one word variable name isn't that good and doesn't tell us enough about that

Reply:
Agreed, done